### PR TITLE
Add unique index to DataFlagOpinions

### PR DESCRIPTION
**Pull request by nritsche**
Adds a unique index combining the columns `type`, `user`, `start_time`, `finish_time`.

### CODE REVIEW

**Comment by nritsche** [comment]
This should really include metadata. But we would have to make that fixed length to be accepted as part of the index and we are not using that metadata yet, anyways.

**Comment by jrs65** [comment]
Can you explain again why you'd want to index on metadata? I would have thought indexing on `type`, `user` `start/finish_time` and `rev` would be sufficient? i.e. a user can only submit one opinion per day per revision of a piece of data.

File: chimedb/dataflag/orm.py [hunk 1]
```diff
@@ -261,6 +261,12 @@ class DataFlagOpinion(DataSubset):
     client = pw.ForeignKeyField(DataFlagClient)
     revision = pw.ForeignKeyField(DataRevision, backref="opinions")
 
+    class Meta:
+        indexes = (
+            # create a unique index
+            (("type", "user", "start_time", "finish_time"), True),
```
Comment (anjakefala) [inline chimedb/dataflag/orm.py]: What will the conflict resolution strategy be?

Comment (jrs65) [inline chimedb/dataflag/orm.py]: I think you should also index on `revision` in here.

**Review by jrs65 — CHANGES_REQUESTED** [review]

**Comment by nritsche** [comment]
> Can you explain again why you'd want to index on metadata? I would have thought indexing on `type`, `user` `start/finish_time` and `rev` would be sufficient? i.e. a user can only submit one opinion per day per revision of a piece of data.

Yes, but the idea of metadata was to later add frequency and input ranges here (see https://github.com/chime-experiment/chimedb_dataflag/blob/f3e8a96c5259bb874bd727b4cb17616baacdfa97/chimedb/dataflag/orm.py#L235).

File: chimedb/dataflag/orm.py (hunk 1 above)
Comment (nritsche) [reply chimedb/dataflag/orm.py]: in bondia the user is warned that they already inserted an opinion, if they click anyways, the `decision` field is updated.

Comment (nritsche) [reply chimedb/dataflag/orm.py]: yes

**Review by jrs65 — APPROVED** [review]

**Comment by nritsche** [comment]
I think I need https://github.com/chime-experiment/chimedb/issues/18 to find out what's wrong here.

File: chimedb/dataflag/orm.py [hunk 2]
```diff
@@ -260,6 +298,14 @@ class DataFlagOpinion(DataSubset):
     last_edit = pw.DoubleField()
     client = pw.ForeignKeyField(DataFlagClient)
     revision = pw.ForeignKeyField(DataRevision, backref="opinions")
+    lsd = pw.IntegerField()
+    note = pw.TextField(null=True)
```
Comment (jrs65) [inline chimedb/dataflag/orm.py]: Everywhere else we call this field `notes`.

**Review by jrs65 — CHANGES_REQUESTED** [review]

**Review by jrs65 — APPROVED** [review]

**Pull request merged**